app/src/main/kotlin/org/kotlinlsp/index/worker/IndexKtFile.kt

```diff
@@ -52,7 +50,117 @@ fun indexKtFile(project: Project, ktFile: KtFile, db: Database) {
+                                Declaration.Class.Type.INTERFACE
+                            } else if (dcl.isAbstract()) {
+                                Declaration.Class.Type.ABSTRACT_CLASS
+                            } else {
+                                Declaration.Class.Type.CLASS
+                            }
+
+                            Declaration.Class(
+                                name,
+                                type,
+                                dcl.fqName?.asString() ?: "",
+                                fileRecord.path,
+                                startOffset,
+                                endOffset
+                            )
+                        }
+
+                        is KtParameter -> {
+                            if (!dcl.hasValOrVar()) return@analyze null
+                            val constructor = dcl.parentOfType<KtPrimaryConstructor>() ?: return@analyze null
+                            val clazz = constructor.parentOfType<KtClass>() ?: return@analyze null
+
+                            Declaration.Field(
+                                name,
+                                dcl.fqName?.asString() ?: "",
+                                fileRecord.path,
+                                startOffset,
+                                endOffset,
+                                dcl.returnType.toString(),
+                                clazz.fqName?.asString() ?: ""
+                            )
+                        }
+
+                        is KtProperty -> {
+                            if (dcl.isLocal) return@analyze null
+                            val clazz = dcl.parentOfType<KtClass>() ?: return@analyze Declaration.Field(
+                                name,
+                                dcl.fqName?.asString() ?: "",
+                                fileRecord.path,
+                                startOffset,
+                                endOffset,
+                                dcl.returnType.toString(),
+                                ""
+                            )
+
+                            Declaration.Field(
+                                name,
+                                dcl.fqName?.asString() ?: "",
+                                fileRecord.path,
+                                startOffset,
+                                endOffset,
+                                dcl.returnType.toString(),
+                                clazz.fqName?.asString() ?: ""
+                            )
+                        }
+
+                        else -> {
+                            // TODO Handle other declarations
+                            warn("Declaration type not handled: ${dcl::class.simpleName}")
+                            null
+                        }
+                    }
+                }
+            } ?: return
+
+            db.putDeclaration(decl)
```

Comment:
In rocksdb we can put multiple items in bulk, here we could do it to improve performance

Reply:
Thinking about it maybe we can leave a `TODO` and do it later, it's just a small improvement